module/VuFind/src/VuFind/Search/Base/Results.php

```diff
@@ -735,4 +755,99 @@ public function getFullFieldFacets(
         } while ($limit == -1 && !empty($facetfields));
         return $facets;
     }
+
+    /**
+     * A helper method that converts the list of facets for the last search from
+     * RecordCollection's facet list.
+     *
+     * @param array $facetList Facet list
+     * @param array $filter    Array of field => on-screen description listing
+     * all of the desired facet fields; set to null to get all configured values.
+     *
+     * @return array Facets data arrays
+     */
+    protected function buildFacetList(array $facetList, array $filter = null): array
+    {
+        // If there is no filter, we'll use all facets as the filter:
+        if (null === $filter) {
+            $filter = $this->getParams()->getFacetConfig();
+        }
+
+        // Start building the facet list:
+        $result = [];
+
+        // Loop through every field returned by the result set
+        $translatedFacets = $this->getOptions()->getTranslatedFacets();
+        $hierarchicalFacets
```

Comment:
If we're moving hierarchical facet processing into the base class, should we just move getHierarchicalFacets up into the base class as well so we can assume it's defined?

Reply:
...and of course this makes me wonder if we should have a separate PR for hierarchical facet adjustments as a prerequisite for this one, to keep all the changes separate.

Reply:
Now in #2397.